core/controllers/admin.py

```diff
@@ -854,3 +854,48 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class UpdateUsernameHandler(base.BaseHandler):
+    """Handler for renaming usernames."""
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        current_username = self.payload.get('current_username', None)
+        new_username = self.payload.get('new_username', None)
+
+        if current_username is None:
+            raise self.InvalidInputException(
+                'Invalid request: The current username must be specified.')
+
+        if new_username is None:
+            raise self.InvalidInputException(
+                'Invalid request: A new username must be specified.')
+
+        if type(current_username).__name__ != 'unicode':
```

Comment:
I tried using `isinstance` but the linter was throwing an error since I was using builtin-unicode. As per Sandeep's suggestion, using python_utils.UNICODE worked.